experimental/traffic-portal/src/app/core/cdns/cdn-table/cdn-table.component.ts

```diff
@@ -0,0 +1,248 @@
+import { ActivatedRoute } from "@angular/router";
+import { ColDef } from "ag-grid-community";
+import { BehaviorSubject } from "rxjs";
+import { AlertLevel, ResponseCDN } from "trafficops-types";
+
+import { CDNService } from "src/app/api";
+import { AlertService } from "src/app/shared/alert/alert.service";
+import { CurrentUserService } from "src/app/shared/current-user/current-user.service";
+import {
+	DecisionDialogComponent,
+	DecisionDialogData
+} from "src/app/shared/dialogs/decision-dialog/decision-dialog.component";
+import type { ContextMenuActionEvent, ContextMenuItem } from "src/app/shared/generic-table/generic-table.component";
+
+/**
+ * CDNTableComponent is the controller for the "CDNs" table.
+ */
+@Component({
+	selector: "tp-cdn-table",
+	styleUrls: ["./cdn-table.component.scss"],
+	templateUrl: "./cdn-table.component.html",
+})
+export class CDNTableComponent implements OnInit {
+	public cdns: Promise<ResponseCDN[]>;
+
+	/* Definitions of the table's columns according to the ag-grid API */
+	public columnDefs: ColDef[] = [
+		{
+			field: "dnssecEnabled",
+			filter: "tpBooleanFilter",
+			headerName: "DNSSEC Enabled",
+			hide: false
+		},
+		{
+			field: "domain",
+			filter: "agTextColumnFilter",
+			headerName: "Domain",
+			hide: false,
+		},
+		{
+			field: "id",
+			filter: "agNumberColumnFilter",
+			headerName: "ID",
+			hide: true,
+		},
+		{
+			field: "lastUpdated",
+			filter: "agDateColumnFilter",
+			headerName: "Last Updated",
+			hide: true
+		},
+		{
+			field: "name",
+			filter: "agTextColumnFilter",
+			headerName: "Name",
+			hide: false,
+		},
+	];
+
+	/**
+	 * Definitions for the context menu items (which act on augmented
+	 * CDN data).
+	 */
+	public contextMenuItems: Array<ContextMenuItem<ResponseCDN>> = [
+		{
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}`,
+			name: "Open in New Tab",
+			newTab: true
+		},
+		{
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}` ,
+			name: "Edit"
+		},
+		{
+			action: "delete",
+			multiRow: false,
+			name: "Delete",
+		},
+		{
+			action: "snapshot-diff",
+			multiRow: false,
+			name: "Diff Snapshot",
+		},
+		{
+			action: "queue",
+			multiRow: false,
+			name: "Queue Server Updates"
+		},
+		{
+			action: "dequeue",
+			multiRow: false,
+			name: "Clear Queued Updates"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/dnssec-keys` ,
+			name: "Manage DNSSEC Keys"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/federations` ,
+			name: "Manage Federations"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/cdns/${selectedRow.id}/delivery-services` ,
+			name: "Manage Delivery Services"
+		},
+		{
+			disabled: (): true => true,
+			href: (selectedRow): string => `/core/profiles?cdnName=${selectedRow.id}` ,
+			name: "Manage Profiles"
+		},
```

Comment:
This is implemented by our current profiles table. I suspect you have it disabled because you couldn't get it to work? The problem is that you can't have query strings in the `href` (maybe I should change the name to `path` because that's all it actually is), you have to use the `queryParams` property of a context menu link.

Reply:
Enabled the *Manage Profiles* link in 29cca13661